### PR TITLE
Add AggreateCounter Controller/Commands

### LINKED ISSUE #426: Add REST-API and shell support for aggregate-counter
As a developer, I'd like to add `aggregate-counter` to artifact registry, so I can use it from shell or the UI/FLO.

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Add list, reset, find operations on Aggregate counter controller
 - Add REST resource support
 - Add tests
 - Add REST client support
 - Add Shell operations/template implementation needed for aggregate-counter commands

This resolves #426

### CODE REVIEW

**Comment by ilayaperumalg** [comment]
This requires #593 and https://github.com/spring-cloud/spring-cloud-stream-app-starters/pull/12

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowControllerAutoConfiguration.java [hunk 1]
```diff
@@ -31,26 +31,8 @@
...
-import org.springframework.cloud.dataflow.server.controller.JobExecutionController;
-import org.springframework.cloud.dataflow.server.controller.JobInstanceController;
-import org.springframework.cloud.dataflow.server.controller.JobStepExecutionController;
-import org.springframework.cloud.dataflow.server.controller.JobStepExecutionProgressController;
-import org.springframework.cloud.dataflow.server.controller.LibraryController;
-import org.springframework.cloud.dataflow.server.controller.ModuleController;
-import org.springframework.cloud.dataflow.server.controller.RestControllerAdvice;
-import org.springframework.cloud.dataflow.server.controller.RootController;
-import org.springframework.cloud.dataflow.server.controller.RuntimeModulesController;
+import org.springframework.cloud.dataflow.server.controller.*;
```
Comment (mbogoevici) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowControllerAutoConfiguration.java]: Expand package.

**Comment by mbogoevici** [comment]
Rebased with some minor polishing here: https://github.com/mbogoevici/spring-cloud-data/commit/5f6540aa78a3d8a53069df1f37bd16c0f5b7a9ab

Since there were some conflicts, some sanity check on whether the functionality is preserved would be good - @ilayaperumalg.

Other than that, LGTM and can be merged.

**Comment by ilayaperumalg** [comment]
@mbogoevici thank you for rebasing and polishing the PR. I will run some sanity checks against it and update here.

**Comment by ilayaperumalg** [comment]
@mbogoevici The changes look good. Also, did some sanity checks on aggregate counter by deploying streams with aggregate counter on `local` dataflow server and the results look good.

**Comment by mbogoevici** [comment]
Squashed and merged w/https://github.com/spring-cloud/spring-cloud-dataflow/commit/5f6540aa78a3d8a53069df1f37bd16c0f5b7a9ab

**Pull request closed without merging**